GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m

```diff
@@ -517,20 +517,13 @@ + (void)proxyRemoteNotificationsMethodsWithAppDelegateSubClass:(Class)appDelegat
       NSSelectorFromString(kGULDidReceiveRemoteNotificationWithCompletionSEL);
   SEL didReceiveRemoteNotificationWithCompletionDonorSEL =
       @selector(application:donor_didReceiveRemoteNotification:fetchCompletionHandler:);
-  if ([appDelegate respondsToSelector:didReceiveRemoteNotificationWithCompletionSEL]) {
```

Comment:
As you mentioned before, it is actually a breaking change. I think we can change the condition to the following:
```
if (![appDelegate didReceiveRemoteNotificationSEL])
```
It is still may be considered as breaking but, in this case the clients that use Auth will get depreciation warning in their code if they still use `didReceiveRemoteNotification:` so it will be more legitimate for the swizzler to force using the new API.

Reply:
I would suggest to add a clear changelog item describing the implications of this change to GoogleUtilities and add a reference to it to Auth to increase visibility of it.

Reply:
What exactly is breaking? We shouldn't do anything that would break a Firebase 7 app.

Reply:
If the method [application:didReceiveRemoteNotification:fetchCompletionHandler:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623013-application?language=objc) is found in the app delegate then the method [application:didReceiveRemoteNotification:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623117-application?language=objc) will never be called.

My suggestion is that we add  [application:didReceiveRemoteNotification:fetchCompletionHandler:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623013-application?language=objc) only if [application:didReceiveRemoteNotification:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623117-application?language=objc) is not found to avoid accidentally breaking an app that still uses the deprecated method. The downside of this approach is that if `application:didReceiveRemoteNotification:` is still used together with Auth, then Auth will never receive the notification. I guess this downside may be mitigated by the changelog and the depreciation warning form on iOS 10+. 

Probably another option would be just hold off on the change until GoogleUtilities stop supporting iOS 9. 

Any thoughts on how to handle it better? Feel free to schedule a chat.

Reply:
It sounds like either way, removing usage of `application:didReceiveRemoteNotification` is a breaking change since some applications may still use it in their own app delegates. I think waiting until the next breaking change release to clean up these usages sounds good.

However, removing the usage of `application:openURL:sourceApplication:annotation:` now should be fine though, correct?

Reply:
@rosalyntan Well, if I understand correctly, it looks pretty similar, if the old method is still used in the app then Auth won't get notified. If so, I would postpone the change until Firebase 8.

Reply:
Gotcha, sounds good to me. I'll wait till Firebase 8 to be on the safe side.